storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -24,149 +26,761 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+const VOLUME_PROTECTION = "storage.metalk8s.scality.com/volume-protection"
+
 var log = logf.Log.WithName("controller_volume")
 
-// Add creates a new Volume Controller and adds it to the Manager. The Manager will set fields on the Controller
-// and Start it when the Manager is Started.
+// Creates a new Volume Controller and adds it to the Manager.
+//
+// The Manager will set fields on the Controller and Start it when the
+// Manager is Started.
 func Add(mgr manager.Manager) error {
 	return add(mgr, newReconciler(mgr))
 }
 
-// newReconciler returns a new reconcile.Reconciler
+// Returns a new reconcile.Reconciler.
 func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 	return &ReconcileVolume{
-		client: mgr.GetClient(),
-		scheme: mgr.GetScheme(),
-		salt:   salt.NewClient(getAuthCredential(mgr.GetConfig())),
+		client:   mgr.GetClient(),
+		scheme:   mgr.GetScheme(),
+		recorder: mgr.GetRecorder("volume_controller"),
+		salt:     salt.NewClient(getAuthCredential(mgr.GetConfig())),
 	}
 }
 
-// add adds a new Controller to mgr with r as the reconcile.Reconciler
+// Adds a new Controller to `mgr` with `r` as the reconcile.Reconciler.
 func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	// Create a new controller
-	c, err := controller.New("volume-controller", mgr, controller.Options{Reconciler: r})
```

Comment:
I'm not fond of these layout and docstring changes. Whilst indeed it may be somewhat easier to read now, they have a massive drawback: when there's a new version of `operator-sdk` and in order to 'upgrade' we again need to recreate/scaffold a project, then `diff` that one with the 'real' code to apply any relevant changes, said diff will be bigger than it needs to be.

Reply:
ha yes, I forgot that we have to do the 'upgrade' by hand/doing a diff…
I'll revert that then.